src/contrib/electron_motion/validateNUE.C

```diff
@@ -0,0 +1,335 @@
+      std::cout<<"costheta = "<<costheta<<" phi = "<<phi<<std::endl;
+      
+
+    }//If is scattering
+
+  }//end event loop
+
+  //TFile *file1 = new TFile(dir+"/Plots/histos.root","recreate");
+  TCanvas canvas("canvas");
+  vx_dist->Draw("HIST");
+  vx_dist->SetLineColor(kBlack);
+  //vx_dist->SetLineWidth
+  vx_dist->SetFillColorAlpha(kBlue,0.35);
+  vx_dist->GetXaxis()->SetTitle("v [c]");
+  vx_dist->SetStats(kFALSE);
+  //vx_dist->Write();
+  vy_dist->Draw("SAME");
+  vy_dist->SetLineColor(kBlack);
+  vy_dist->SetFillColorAlpha(kRed,0.35);
+  //vy_dist->Write();
+  vz_dist->Draw("SAME");
+  vz_dist->SetLineColor(kBlack);
+  vz_dist->SetFillColorAlpha(kGreen,0.35);
+  //vz_dist->Write();
+  //file1->Close();
+  auto legend = new TLegend(0.1,0.7,0.3,0.9);
+  //legend->SetHeader("The Legend Title","C"); // option "C" allows to center the header
+  legend->AddEntry(vx_dist,"vx");
+  legend->AddEntry(vy_dist,"vy");
+  legend->AddEntry(vz_dist,"vz");
+  legend->Draw();
+
+  std::string file_out = "Plots/vdist.pdf";
+  const char *fo = file_out.c_str();
+
+  canvas.Print(fo);
+
+  TCanvas canvas2("canvas2");
+  ang_dist->Draw("COLZ");
+  ang_dist->GetXaxis()->SetTitle("cos#theta");
+  ang_dist->GetYaxis()->SetTitle("#phi");
+  ang_dist->SetStats(kFALSE);
+  std::string file_out2 = "Plots/angdist.pdf";
+  const char *fo2 = file_out2.c_str();
+  canvas2.Print(fo2);
+}
+
+void make_y_dist(TString filename = "gntp.0.ghep.root", TString dir=""){
+  TFile infile(dir+filename);
+  //TFile infile('gntp.0.ghep.root');
+
+  // Get the GENIE GHEP tree and set its branch address
+  TTree * tree = dynamic_cast<TTree*> ( infile.Get("gtree") );
+
+  //SEt histograms
+  TH1F *y_dist = new TH1F("y","y",20,0,1);
+
+  NtpMCEventRecord * mcrec = 0;
+  tree->SetBranchAddress( "gmcrec", &mcrec);
+
+  // Event loop
+  for(Long64_t i=0; i < tree->GetEntries(); i++){
+    tree->GetEntry(i);
+
+    EventRecord & event = *( mcrec->event );
+    
+    const Interaction & inter = *( event.Summary() ) ;
+
+    const ProcessInfo & proc_info = inter.ProcInfo() ;
+
+    if (proc_info.IsElectronScattering()){
+      GHepParticle * electron = event.HitElectron(); //Find initial electron
+      const TLorentzVector & pe = *( electron -> P4() ) ; //Get its 4 momentum
+
+      //Fill histograms
+      y_dist->Fill(inter.Kine().y(true));
+
+    }//If is scattering
+
+  }//end event loop
+
+  TCanvas canvas("canvas");
+  y_dist->Draw("HIST");
+  y_dist->SetLineColor(kBlack);
+  y_dist->SetFillColorAlpha(kBlue,0.85);
+  y_dist->GetXaxis()->SetTitle("y");
+  y_dist->SetStats(kFALSE);
+  std::string file_out = "Plots/ydist.pdf";
+  const char *fo = file_out.c_str();
+  canvas.Print(fo);
+}
+
+void validateNUE(TString filename = "gntp.0.ghep.root",TString dir=""){
+  //Make all distributions
+  gSystem->Exec("mkdir -p Plots");
+  make_vdist(filename,dir);
+  make_y_dist(filename,dir);
+}
+
+void make_finale_tree(TString filename = "gntp.0.ghep.root", TString dir=""){
```

Comment:
I'm worried that here we are reinventing the wheel, what is this doing differently other than the standard gst format?